### PR TITLE
Containing all bokeh style to .bk-plot

### PR DESCRIPTION
**Pull request by mistakevin**
This will help better contain all of bokeh's style inside the `.bk-plot` class. There is bleeding regarding global html tags that the `.bk-*` naming convention just won't fix.

See here the `<label></label>` tag is inheriting bokeh's style
screen shot 2016-02-01 at 9 48 39 am

This PR will contain all styles to inside the `.bk-plot` class

screen shot 2016-02-01 at 9 48 59 am

screen shot 2016-02-01 at 9 50 53 am

See https://github.com/bokeh/bokeh/issues/2589

### CODE REVIEW

**Comment by mistakevin** [comment]
@bokeh/dev

**Comment by mistakevin** [comment]
@mattpap I chose `.bk-plot` because assumed all plots were already wrapped in this class. If we add a new class, are we going to auto generate the templates with the new class? If so, where would you suggest putting that in? The plot style will break if it is not wrapped, so it can't be optional.

**Comment by mattpap** [comment]
`bokeh.embed` will have to be changed accordingly, so that `bk-bokeh` is added along `plotdiv`. Then `plotdiv` should be deprecated as it doesn't have prefix.

**Comment by birdsarah** [comment]
The integration test failures on this PR are real - the selenium tests use css selectors to find elements and I'm guessing that this change has broken them.

Also, isn't `bk-bokeh` a little cumbersome? I thought `bk` was an abbreviation of `bokeh` so it seems a bit redundant. I would say just `bokeh` or `bk` is fine - my preference would be for `bk` as it makes the sub levels like `bk-plot` make more sense.

**Comment by mattpap** [comment]
Well, we can call it `bk-root` or whatever. The point is to have `bk-` prefix consistently everywhere. Note that `bk-` prefix should be configurable if it causes conflicts with some environment (not that we are even close to making this possible).

**Comment by mattpap** [comment]
> The integration test failures on this PR are real

Perhaps that's true, but they fail in very unhelpful way: `TypeError: string indices must be integers`.

**Comment by mattpap** [comment]
Also, reviewing this PR, I don't see anything that could interfere with selenium.

**Comment by mattpap** [comment]
OK, most likely we do embedding improperly in some/all cases. I recall not seeing `plotdiv`, so `bk-bokeh` isn't going to be there as well and then no CSS will be applied.

**Comment by birdsarah** [comment]
> Perhaps that's true, but they fail in a very unhelpful way: TypeError: string indices must be integers.

Agreed - which is why I'm getting the better HTML reporting set-up. So you have an easy link to a video on failure.

**Comment by mattpap** [comment]
@birdsarah, in this case it's more about a robust implementation (e.g. JSON schema validation) than nicer error reporting.

**Comment by birdsarah** [comment]
@mattpap, can you clarify? When you write:

>  it's more about a robust implementation (e.g. JSON schema validation) than nicer error reporting

It reads to me like you're suggesting fixing the guts of selenium. This does not seem like a reasonable suggestion.

**Comment by mattpap** [comment]
@birdsarah, I thought this was our code. I don't expect we will be fixing selenium, thought this piece of code in the tracebace should be rewritten from scratch.

**Comment by birdsarah** [comment]
> @birdsarah, I thought this was our code. I don't expect we will be fixing selenium, thought this piece of code in the tracebace should be rewritten from scratch.

And just confirming, that given that the code in the traceback isn't ours, you're ok with me working on trying to give more helpful feedback given selenium's limitations.

**Comment by mattpap** [comment]
@birdsarah, obviously yes.

**Comment by birdsarah** [comment]
@mistakevin would you like me to push fixes to the selenium tests to this branch?

**Comment by mistakevin** [comment]
@birdsarah Yes!

**Comment by birdsarah** [comment]
I was mistaken, the selenium failure, was not a result of this work, but of the selenium window being too small and the box_select not being on-page to click.

This is fixed in PR #3896. 

When running this locally, I am having a problem with the sliders app. It appears that something is now missing from bokeh-widgets.css.

Here's what it looks like:

screen shot 2016-02-23 at 6 47 03 pm

This is what master looks like:

screen shot 2016-02-23 at 6 47 35 pm

I may have compiled something wrong, I just ran the regular `python setup.py develop --build-js` and then did `../../bin/bokeh serve --show sliders.py` from `examples/apps`

File: bokehjs/src/less/bokeh-widgets.less [hunk 1]
```diff
@@ -1,3 +1,5 @@
-@import "vendor.less";
-@import "overrides.less";
-@import "widgets.less";
+.bk-plot {
```
Comment (birdsarah) [inline bokehjs/src/less/bokeh-widgets.less]: I believe this is the source of my sliders problem, the nesting isn't working as expected. My `bokeh-widgets.css` output in `bokehjs/build/css`

starts:
```css
.bk-plot {
  /* makes the font 33% larger relative to the icon container */
  /* Font Awesome uses the Unicode Private Use Area (PUA) to ensure screen
   readers do not read off random characters that represent icons */
  /*!
 * jQuery UI CSS Framework 1.11.2
 * http://jqueryui.com
 *
 * Copyright 2014 jQuery Foundation and other contributors
 * Released under the MIT license.
 * http://jquery.org/license
 *
 * http://api.jqueryui.com/category/theming/
 */

/* Layout helpers
----------------------------------*/
.bk-ui-helper-hidden {
	display: none;
}
.bk-ui-helper-hidden-accessible {
```

Note the nesting - this doesn't work in css only in less or sass. Not sure quite what the problem is.

**Comment by mistakevin** [comment]
@birdsarah Good catch. Looks like we were explicitly bringing in the CSS files as `inline`. Treating them as LESS seemed to do the trick.

File: bokehjs/src/vendor/slick-grid-2.1.0/slick-default-theme.css [hunk 2]
```diff
@@ -96,7 +96,6 @@ classes should alter those!
 
 .bk-slick-row.loading {
   opacity: 0.5;
-  filter: alpha(opacity = 50);
```
Comment (mistakevin) [inline bokehjs/src/vendor/slick-grid-2.1.0/slick-default-theme.css]: This is for pre-IE8. Hoping we can live without it, as LESS does not support it.

Comment (bryevdv) [inline bokehjs/src/vendor/slick-grid-2.1.0/slick-default-theme.css]: We don't support even IE8, it does't have typed arrays.

**Comment by birdsarah** [comment]
Have re-tested this, seems to be working fine.

**Comment by canavandl** [comment]
I'm unable to get the tool icons to load using this branch:

screen shot 2016-02-24 at 2 38 46 pm

**Comment by canavandl** [comment]
I think it's an issue of bringing in the the css as less (the icon pngs are generated css from a sphinx static folder)

**Comment by canavandl** [comment]
The icon problems appears to be a defect in the `relative-dev` and `absolute-dev` BOKEH_RESOURCES settings that is in master and not related to this PR. 

So this LGTM and i'll open a hotfix PR for the resources issue.

**Comment by mistakevin** [comment]
@mattpap After merging this, I just realized I accidently reverted back to `.bk-plot` after we had decided to go with something else. If this is an issue, I'll be happy to write-up a small PR changing it.

cc. @birdsarah @canavandl @bryevdv

**Pull request merged**